### PR TITLE
s3 data validation: annotations

### PR DESCRIPTION
**Pull request by daniel-ji**
Add validation tests for annotations, some new, most of them ported over & refactored from Utz's @uermel tests. For a list of tests, see https://docs.google.com/document/d/1yMKM0DW9KRhlcYiBGPcR7oW0liGtUew6NAmBhMg5U3w/edit#heading=h.56ec9se80d6g

Also refactored data / file fetching to reflect that there is only one mrc / zarr file per segmentation mask annotation / tomogram / tiltseries. Note that in the metadata files, the `mrc_files` field in the path still is an list of strings and should get changed to be a single string (not done in this PR).

### CODE REVIEW

File: ingestion_tools/scripts/data_validation/fixtures/data.py [hunk 1]
```diff
@@ -18,24 +18,34 @@
...
+    with fs.open(zarrfile + "/.zattrs", "r") as f:
+        zattrs = json.load(f)
+    return zattrs
+
+
+def get_zarrays(zarrfile: str, fs: FileSystemApi) -> Dict:
+    """Get the zarray for a zarr volume file without downloading the entire file."""
+    zarrays = {}
+    expected_subfolders = [f"{zarrfile}/{i}" for i in range(3)]
+    actual_subfolders = ["s3://" + folder for folder in fs.glob(zarrfile + "/*/")]
```
Comment (jgadling) [inline ingestion_tools/scripts/data_validation/fixtures/data.py:42]: This is fine, but we can also use the omezarr reader to do the same thing. I think we should start thinking about whether we want to use filesystem checks vs omezarr reader checks more consistently.

File: ingestion_tools/scripts/data_validation/fixtures/data.py
Comment (daniel-ji) [inline ingestion_tools/scripts/data_validation/fixtures/data.py:1]: The main changes in this file:
- changing the code to handle one mrc / zarr file, not multiple mrc / zarr files
- simplifying some fixtures' dictionary structure (no need to have nested dictionary structures in some cases)
- adding zarr header retrieval
- improving type annotations / documentation

File: ingestion_tools/scripts/data_validation/fixtures/path.py
Comment (daniel-ji) [inline ingestion_tools/scripts/data_validation/fixtures/path.py:1]: Changes here:
- Remove accommodation for multiple mrc files per seg mask annotation / tiltseries / tomogram
- Improve variable naming / typing

**Review by jgadling — APPROVED** [review]

**Comment by daniel-ji** [comment]
Closed, a later s3 data validation PR affected a lot of pieces this PR was intending to merge in, so will create a new refined PR

**Pull request closed without merging**